### PR TITLE
Updates to documentation for root package and tf.a*

### PR DESCRIPTION
**Pull request by frreiss**
This PR contains miscellaneous API documentation updates that cover the root package `tf` and all the API functions whose names start with `tf.a`. Some of the changes correct typos, while others document aspects of these functions that weren't obvious to me without reading the source code.

Note that the functions `tf.abs`, `tf.add_n`, `tf.accumulate_n`, and `tf.angle` moved to the `tf.math` package while I was making these edits.

I've generated and reviewed the Markdown files for the documentation after these changes.

### CODE REVIEW

File: tensorflow/python/ops/math_ops.py [hunk 1]
```diff
@@ -568,8 +572,8 @@ def angle(input, name=None):
   For example:
 
   ```
-  # tensor 'input' is [-2.25 + 4.75j, 3.25 + 5.75j]
-  tf.angle(input) ==> [2.0132, 1.056]
+  # tensor 'input' is [-2.25 + 4.75j, 3.25 + 5.75j], in `complex64` format
```
Comment (MarkDaoust) [inline tensorflow/python/ops/math_ops.py]: To make it one step closer to runnable code you could say:

```suggestion
input = tf.constant([-2.25 + 4.75j, 3.25 + 5.75j])
```

File: tensorflow/python/ops/math_ops.py [hunk 2]
```diff
@@ -568,8 +572,8 @@ def angle(input, name=None):
...
+  tf.angle(input) ==> array([2.0131705, 1.056345 ], dtype=float32)
```
Comment (MarkDaoust) [inline tensorflow/python/ops/math_ops.py]: And convert this to a comment (I get `float64`).

```suggestion
  tf.angle(input).numpy()
  # ==>  array([2.0131705, 1.056345 ], dtype=float64)
```

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
Thanks for all these fixes.

Here are a couple of tiny suggestions.

I cross-checked everything I could, and everything looks right.

The only thing I have no context on is `gradients_impl.py`. But it does _sound_ right.

File: tensorflow/python/ops/math_ops.py (hunk 1 above)
Comment (frreiss) [reply tensorflow/python/ops/math_ops.py]: Good idea, change applied.

File: tensorflow/python/ops/math_ops.py (hunk 2 above)
Comment (frreiss) [reply tensorflow/python/ops/math_ops.py]: Depending on the dtype the constant `input`, I either get `array([2.01317055, 1.05634501])` (`input` is of dtype `complex128`) or `array([2.0131705, 1.056345 ], dtype=float32)` (`input` is of dtype `complex64`). The commit I just pushed has a matching set of runnable code and exact output for the complex64 case.

**Comment by frreiss** [comment]
It looks like there was a hiccup somewhere in the CI infrastructure. The build and tests are all working on this branch on my test machine. Would someone with access mind rerunning the CI checks for this PR?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by ymodak** [comment]
@frreiss Can you please resolve the conflicts? Thanks!

**Comment by frreiss** [comment]
Merge conflicts fixed. Currently rerunning regression tests on my test machine.

**Comment by ymodak** [comment]
@frreiss There are branch conflicts. Can you please rebase? Thanks!

**Comment by frreiss** [comment]
Merged in changes from master, including manually moving some of my documentation edits from `gradients_impl.py` to `gradients_util.py`. The branch conflicts should be fixed now.

**Comment by ymodak** [comment]
@MarkDaoust Can you PTAL? Thanks!

**Comment by frreiss** [comment]
Hi @MarkDaoust and @ymodak , this PR appears to be ready to merge. Is there anything still holding it up?

**Comment by rthadur** [comment]
@frreiss can you please resolve conflicts

**Comment by frreiss** [comment]
Conflicts resolved.

**Review by robieta — APPROVED** [review]
This is really lovely. Thanks so much for putting this together.

(I checked gradients_impl.py, and it looks good.)

**Pull request merged**